src/main/kotlin/near/vanilla/death/database/DeathDatabase.kt

```diff
@@ -0,0 +1,136 @@
+package near.vanilla.death.database
+
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import org.bukkit.Bukkit
+import org.bukkit.entity.Player
+import org.bukkit.event.entity.PlayerDeathEvent
+import org.bukkit.inventory.ItemStack
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.transactions.transaction
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.*
+
+
+object DeathDatabase {
+
+    private val dataSource : HikariDataSource
+
+    private object TAGS {
+        val driverClassName = "driverClassName"
+        val jdbcUrl = "jdbcUrl"
+        val user = "username"
+        val password = "password"
+        val cachePrepStmts = "dataSource.cachePrepStmts"
+        val prepStmtCacheSize = "dataSource.prepStmtCacheSize"
+        val prepStmtCacheSqlLimit = "dataSource.prepStmtCacheSqlLimit"
+    }
+
+    init {
+        val config = HikariConfig(getProperties().absolutePath)
+        transaction(
+                Database.connect(url = config.jdbcUrl,
+                        driver = config.driverClassName,
+                        user = config.username,
+                        password = config.password)) {
+            SchemaUtils.createDatabase("deathdatabase")
+        }
+        dataSource = HikariDataSource(config)
+        transaction(Database.connect(dataSource)) {
+            SchemaUtils.create(DeadPlayers)
+        }
+    }
+
+    operator fun get(player: Player) : List<ItemStack> {
+        // Thread safe by setting val
+        val jsonItems = this[player.uniqueId] ?: return emptyList()
+        return ItemSerializer.deserialize(jsonItems)
+    }
+
+    fun set(playerDeathEvent: PlayerDeathEvent) = set(playerDeathEvent.entity, playerDeathEvent.drops)
+
+    operator fun set(player: Player, items : List<ItemStack>) = set(player.uniqueId, ItemSerializer.serialize(items))
+
+    private operator fun get(uuid : UUID) : String? {
+        var result : String? = ""

```

Comment:
Shouldn't this be null? RN there's 3 possible return values - legitimate, empty string and null.
```suggestion
        var result : String? = null
```